src/main/scala/commonist/util/OaiPmh2.scala

```diff
@@ -0,0 +1,122 @@
+package commonist.util
+
+import commonist.data._
+
+import scala.xml._
+
+import scutil.log._
+
+/** 
+ * Parse and extract image metadata from OAI-PMH 2.0 files.
+ * See www.openarchives.org/OAI/openarchivesprotocol.html
+ */
+class OaiPmh2(doc:Elem,props:Map[String,String]) extends Logging {
+
+  /** fix badly formatted XML files (escaped twice) **/
+  private def text(e:NodeSeq) = {
+    e.map(_.text).mkString("\n").trim()
+     .replaceAll("&amp;", "&").replaceAll("&quot;", "\"").replaceAll("&apos;", "'").replaceAll("&lt;", "<").replaceAll("&gt;", ">")
+  }
+
+  private def artist(creator:String):String = {
+    val prop = creator.replaceAll(" ", "_")
+    if (props.contains(prop))
+      return props(prop)
+    return creator
+  }
+
+  private def permission(rights:String):String = {
+    val lowercase = rights.toLowerCase()
+    if (lowercase.contains("public domain")
+     || lowercase.contains("domaine public")
+     || lowercase.contains("gemeinfreiheit")
+     || lowercase.contains("dominio público")
+     )
+      return "{{PD-old|PD-70}}"
+    return rights
+  }
+
+  private def medium(format:String):String = {
+    val lowercase = format.toLowerCase()
+    if (lowercase.contains("photo") || lowercase.contains("foto"))
+      return "{{Technique|photograph}}"
+    return format
+  }
+
+  private def dimensions(format:String):String = {
+    ".*; (\\d+) x (\\d+) cm ;.*".r.findAllMatchIn(format).foreach { m =>
+      return "{{Size|unit=cm|height=" + m.group(1) + "|width=" + m.group(2) + "}}"
+    }
+    return ""
+  }
+  
+  private def description(fullDescription:String,historyIndex:Int):String = {
```

Comment:
lacks a blank before the second argument - and what exactly does the historyIndex mean - can we have something stronger typed, and do we really need that?

Reply:
I have removed this method.